### PR TITLE
fix(#2028): add accessibility features

### PR DESCRIPTION
**Pull request by syedszeeshan**
# Before (the change)

# After (the change)
-The code adds appropriate aria-* attributes based on whether a field is required, has helper text and/or error text or both.
-When focused, the screen reader announces the helper text and/or error text

## Make sure that you've checked the boxes below before you submit the PR

- [x] I have read and followed the [setup steps](https://goa-dio.atlassian.net/wiki/spaces/DS/pages/3086385489/Contribution+-+web+content#%5BinlineExtension%5DCode-contributions)
- [x] I have created necessary unit tests
- [ ] I have tested the functionality in both React and Angular.

## Steps needed to test

**Playground Test Code**

```
import { Component } from "@angular/core";
import { FormBuilder, FormControl, FormGroup, Validators } from "@angular/forms";

@Component({
  selector: "abgov-radio",
  templateUrl: "./radio.component.html",
  styleUrls: ["./radio.component.css"],
})
export class RadioComponent {
  form: FormGroup;
  inputFormCtrl = new FormControl("");
  radioFormCtrl = new FormControl("");
  chkbox = new FormControl();
  txtArea = new FormControl();
  colorDropdown = new FormControl();

  constructor(private fb: FormBuilder) {
    this.form = new FormGroup({
      inputFormCtrl: this.inputFormCtrl,
      radioFormCtrl: this.radioFormCtrl,
      chkbox: this.chkbox,
      txtArea: this.txtArea,
      colorDropdown: this.colorDropdown,
    });
  }

  onSubmit() {
    this.inputFormCtrl.setValidators([Validators.required]);
    this.inputFormCtrl.updateValueAndValidity();

    this.radioFormCtrl.setValidators([Validators.required]);
    this.radioFormCtrl.updateValueAndValidity();

    this.chkbox.setValidators([Validators.required]);
    this.chkbox.updateValueAndValidity();

    this.txtArea.setValidators([Validators.required]);
    this.txtArea.updateValueAndValidity();

    this.colorDropdown.setValidators([Validators.required]);
    this.colorDropdown.updateValueAndValidity();

    this.form.markAllAsTouched();

    if (this.form.valid) {
      console.log("Form is valid");
    }
  }
}
```

```
<form [formGroup]="form" style="margin-top: 100px;">
<goa-form-item
        label="Amount"
        requirement="required"
        helptext="Help Text: This is the amount you want to transfer."
        [error]="form.controls['inputFormCtrl'].invalid && form.controls['inputFormCtrl'].touched ? 'Error Message: You must enter an amount.' : null">

  <goa-input
          name="inputFormCtrl"
          goaValue
          width="398px"
          [formControl]="inputFormCtrl"
          [error]="form.controls['inputFormCtrl'].invalid && form.controls['inputFormCtrl'].touched"
  ></goa-input>
</goa-form-item>

<br>

<goa-form-item
    label="Accounts"
    mb="l"
    helptext="Help Text: Select an account"
    [error]="form.controls['radioFormCtrl'].invalid && form.controls['radioFormCtrl'].touched ? 'Error Message: You must select an account.' : null"
>
<goa-radio-group
    goaValue
    formControlName="radioFormCtrl"
    [error]="form.controls['radioFormCtrl'].invalid && form.controls['radioFormCtrl'].touched"
    >
  <goa-radio-item value="1" label="Account 1"></goa-radio-item>
  <goa-radio-item value="2" label="Account 2"></goa-radio-item>
  <goa-radio-item value="3" label="Account 3"></goa-radio-item>
</goa-radio-group>
</goa-form-item>

<br>

<goa-form-item
  label="Desserts"
  mb="l"
  helptext="Help Text: Select a dessert"
  [error]="form.controls['chkbox'].invalid && form.controls['chkbox'].touched ? 'Error Message: You must select the ice cream.' : null">
  <goa-checkbox
    name="desserts"
    text="Ice Cream"
    value="10"
    goaChecked
    [formControl]="chkbox"
    [error]="form.controls['chkbox'].invalid && form.controls['chkbox'].touched"
  >
</goa-checkbox>
</goa-form-item>


<br>
<goa-form-item
  label="Charges"
  [error]="form.controls['txtArea'].invalid && form.controls['txtArea'].touched ? 'Error Message: You must enter some details.' : null"
  helpText="Help Text: Enter the charges">
  <goa-textarea goaValue
      [formControl]="txtArea" width="944px" countby="character" maxcount="500" countby="character"
      [error]="form.controls['txtArea'].invalid && form.controls['txtArea'].touched" />
</goa-form-item>

<br>
<goa-form-item
  requirement="required"
  mt="l"
  label="Colors Dropdown"
  helpText="Help Text: Select a color from the dropdown list"
  [error]="form.controls['colorDropdown'].invalid && form.controls['colorDropdown'].touched ? 'Error Message: You must pick a color.' : null">
<goa-dropdown [formControl]="colorDropdown" goaValue name="item" value="" width="200px"
              [error]="form.controls['colorDropdown'].invalid && form.controls['colorDropdown'].touched">
  <goa-dropdown-item value="red" label="Red"></goa-dropdown-item>
  <goa-dropdown-item value="green" label="Green"></goa-dropdown-item>
  <goa-dropdown-item value="blue" label="Blue"></goa-dropdown-item>
</goa-dropdown>
</goa-form-item>

</form>
<br><br>
<goa-button (_click)="onSubmit()" type="primary">Submit</goa-button>

```

### CODE REVIEW

**Comment by lizhuomeng71** [comment]
Helper Text and Error Text Was not read out by NVDA and Narrator

The Requirement of this ticket specified that help text and error text should be ready out, however, even aria-describedby="helptext-akr95iq" exist, NVDA and Narrator does not read it out

Code Example as such

<goa-form-item label="Input Label" requirement="required" labelsize="regular" helptext="This is Basic Helper Text" error="This is Basic Error Text">
  <goa-input width="20ch" error="true" name="item" [value]="value" (_change)="onChange($event)"></goa-input>
</goa-form-item>

**Comment by syedszeeshan** [comment]
https://github.com/user-attachments/assets/e5c552a8-2da1-4e49-a956-fafddcb7a718

**Comment by syedszeeshan** [comment]
https://github.com/user-attachments/assets/9a16ed3a-7508-4397-b33a-6a5dea3d74aa

**Comment by syedszeeshan** [comment]
https://github.com/user-attachments/assets/a9b0adaf-4c66-467b-913d-f23e9de18863

**Comment by syedszeeshan** [comment]
https://github.com/user-attachments/assets/04fd8e1b-fc57-4a45-9ada-9a53f2abac31

**Comment by lizhuomeng71** [comment]
> But when there is validation error, and an input element is focused, the screen reader announces the ~helper text~ error text.

For this case, the NVDA still announces the ~helper text~ in the audio (EVEN the print to screen show it announce the error text)

File: libs/web-components/src/components/checkbox/Checkbox.svelte [hunk 1]
```diff
@@ -32,22 +46,31 @@
...
+    if (isError !== prevError) {
+      dispatch(_rootEl, "errorChange", { isError }, { bubbles: true });
+      prevError = isError;
+    }
+  }
   $: isChecked = toBoolean(checked);
   $: isIndeterminate = false; // Design review. To be built with TreeView Later
 
-  onMount(() => {
+  onMount(async () => {
```
[the remark this answers is not in the document]
Comment (syedszeeshan) [reply libs/web-components/src/components/checkbox/Checkbox.svelte]: PR has been updated.

File: libs/web-components/src/components/dropdown/Dropdown.svelte [hunk 2]
```diff
@@ -74,13 +74,14 @@
 
   let _mountStatus: "active" | "ready" = "ready";
   let _mountTimeoutId: any = undefined;
+  let isError = toBoolean(error);
```
[the remark this answers is not in the document]
Comment (syedszeeshan) [reply libs/web-components/src/components/dropdown/Dropdown.svelte]: PR has been updated.

File: libs/web-components/src/components/form-item/FormItem.svelte [hunk 3]
```diff
@@ -46,9 +56,13 @@
   export let error: string = "";
   export let requirement: RequirementType = "";
   export let maxwidth: string = "none";
-  export let id: string = "";
+  export let id: string = ""; // @deprecated: no longer used
 
   let _rootEl: HTMLElement;
+  let inputEl: HTMLElement;
```
[the remark this answers is not in the document]
Comment (syedszeeshan) [reply libs/web-components/src/components/form-item/FormItem.svelte]: PR has been updated.

File: libs/web-components/src/components/form-item/FormItem.svelte [hunk 4]
```diff
@@ -118,7 +201,8 @@
   </div>
 
   {#if $$slots.error || error}
-    <div class="error-msg">
+    <!-- TODO: polite or assertive ?? -->
```
[the remark this answers is not in the document]
Comment (syedszeeshan) [reply libs/web-components/src/components/form-item/FormItem.svelte]: The comment has been removed, PR updated. 

In this case, If I remove the role alert, the announcements stop working when there is an error.

File: libs/web-components/src/components/input/Input.svelte [hunk 5]
```diff
@@ -89,8 +97,14 @@
   $: handlesTrailingIconClick = toBoolean(handletrailingiconclick);
   $: isFocused = toBoolean(focused);
   $: isReadonly = toBoolean(readonly);
-  $: isError = toBoolean(error);
   $: isDisabled = toBoolean(disabled);
+  $: {
```
[the remark this answers is not in the document]
Comment (syedszeeshan) [reply libs/web-components/src/components/input/Input.svelte:101]: PR updated

File: libs/web-components/src/components/form-item/FormItem.svelte [hunk 6]
```diff
@@ -69,8 +83,79 @@
...
+    if (_hasError !== ce.detail.isError) {
+      _hasError = ce.detail.isError;
+      updateAriaDescribedBy();
+    }
+  }
+
+  function handleAnnounceHelperText() {
+    const message = _hasError ? error || helptext : helptext;
+    if (message) {
+      console.log(
```
[the remark this answers is not in the document]
Comment (syedszeeshan) [reply libs/web-components/src/components/form-item/FormItem.svelte]: yes, PR has been updated now.

File: libs/web-components/src/components/form-item/FormItem.svelte [hunk 7]
```diff
@@ -46,9 +56,13 @@
   export let error: string = "";
   export let requirement: RequirementType = "";
   export let maxwidth: string = "none";
-  export let id: string = "";
+  export let id: string = ""; // @deprecated: no longer used
```
Comment (ArakTaiRoth) [inline libs/web-components/src/components/form-item/FormItem.svelte:59]: If this is being deprecated, people should be made aware of that. In react-components check out line 38 in `dropdown-item.tsx`, and for web-components look at lines 60 and 66 in `IconButton.svelte` for an idea on how to do that.

Comment (chrisolsen) [reply libs/web-components/src/components/form-item/FormItem.svelte:59]: I don't think `id` was actually used within the FormItem.svelte component, so in this case I don't think any warnings are necessary with the svelte file. But it should also be deprecated on the form-item.tsx file.

Comment (syedszeeshan) [reply libs/web-components/src/components/form-item/FormItem.svelte]: I have removed this comment` // @deprecated: no longer used`.... I checked the alpha branch and it has been removed in later revisions to this file. When I started working on this ticket, the comment was there, but not anymore.

**Pull request merged**